src/stdlib_strings_format_string.fypp

```diff
@@ -0,0 +1,52 @@
+#:include "common.fypp"
+#:set RIL_KINDS_TYPES = REAL_KINDS_TYPES + INT_KINDS_TYPES + LOG_KINDS_TYPES
+submodule (stdlib_strings) stdlib_strings_format_string
+
+    implicit none
+    integer, parameter :: buffer_len = 512
+
+contains
+
+    #:for kind, type in RIL_KINDS_TYPES
+    module procedure format_string_${type[0]}$${kind}$
+        !! Format ${type}$ variable as character sequence
+        character(len=buffer_len) :: buffer
+        integer :: stat
+        if(present(fmt)) then
```

Comment:
maybe optval is preferred, i.e. `write(buffer, optval(fmt, default_fmt), iostat=stat) val`

Reply:
`optval` might work if we pass `(g0)` as `default_fmt`, which should be pretty much what we get from list-directed IO.